src/web/services/deadline-extension-helper.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Deadline Extension utility functions.
+ */
+export class DeadlineExtensionHelper {
+  public static hasOngoingExtension(deadlines: {
+    studentDeadlines: Record<string, number>,
+    instructorDeadlines: Record<string, number>,
+  }): boolean {
+    const timeNow = Date.now();
+    const allDeadlines: Record<string, number> = { ...deadlines.studentDeadlines, ...deadlines.instructorDeadlines };
```

Comment:
Possible bug here. Suppose the following deadlines

```
studentDeadlines = {"abc@gmail.com": futureTime};
instructorDeadlines = {"abc@gmail.com": pastTime};
```

In this case, I believe the `abc@gmail.com`  in `instructorDeadlines` will override the one in `studentDeadlines` and give incorrect output.

Probably need to check both separately to be safe.